scripts/parse-vcf/README.md

```diff
@@ -0,0 +1,40 @@
+
+Though the `docker run...` approach will work, it is less than ideal given the program's current state. Because `parse_vcf` uses the python multiprocessing module to spawn child processes, these processes won't be able to print their progress/debugging messages to the host machine. It will likely be easier to run and debug the script from within the container itself, either by opening it in vscode or logging in with something like `docker run -it --rm -v `pwd`:`pwd`-w`pwd` --entrypoint="" bcftools bash`.
+
+## What the script does
+
+The parsing script will process a vcf of variant annotations, returning a subset of fields and combining them into a single csv, which can then be moved to a database. By filtering out unnecessary values, the size of the annotation file can be greatly reduced. Besides the `pos`, `chrom`, `alt`, and `ref` fields, the script will also return `nhomalt`, `AF`, `AC` and relevant `VEP` fields (if available). If you would like to adjust this output, the script will have to be manually updated (see `parse_vcf_response` function). There is also some data type coercion that is tailored to the configuration of the mongo database for ease of import that may need to be changed if the script is used for other purposes.
+
+To improve efficiency, the script uses the python multiprocessing module to process chunks of the vcf file in parallel. It does this by reading the file's header and breaking each chromosome into a collection of subregions, each of which is passed into a processing queue. The script then deploys workers (the default number is 2, but this can be changed with the `--concurrency_max` argument) to the queue for parallel processing. Once the queue is exhausted, the script resumes execution, repeating the same process on the next chromosome until there are no more annotations to process. The results are written to a csv in the directory speficied by the `--out` argument. Performance will vary depending on resources, and it is usually necessary to try a variety of combinations for the `--chunk_size` and `--concurrency_max` arguments before finding an optimal configuration.
+
+## Limitations
+
+The `parse_vcf` module is very much a work in progress and currently has many limitations.
+
+- If execution is interrupted, it will be difficult to restart. While passing a region to the `--start-position` argument will begin the script at a given locus, this is not guarnteed to restart the script where it left off. Because jobs are handled in parallel and, moreover, failed jobs are pushed back into the queue, care will have to be taken to determine which regions remain to be processed when examining an incomplete results file. Also, bear in mind that the script will not remove a csv file that already exists, so if you are restarting a script from scratch after an error, be sure to manually remove the results file first (or point to a new one), otherwise the results will be appended to the extant file.
```

Comment:
This might be a silly question, but say we are using `parse_vcf` for a large VCF and somewhere along the line, the script fails due to some errors that we did not foresee, we'd have to delete the CSV file that existed and rerun the whole process again?

Reply:
Yes, this is the sad truth. I think that to reliably restart the process, there would need to be some extra work done to parse the csv, compare with what's in the vcf headers, determine the missing rows, and relaunch the parsing script accordingly (would probably need to update the script to take config of many regions).

Reply:
Right, that's what I thought too. Thanks for clarifying!